File: implementations/node/sdk/src/contracts/observation/execution-observation.ts
import { AI_SDK_SCHEMA_VERSIONS } from "../common/schema-versions.js";
import type { AiSdkExecutionStatus } from "../executions/execution-status.js";
import type { AiSdkExecutionStepObservation } from "./execution-step-observation.js";

export interface AiSdkExecutionObservation {
  readonly schemaVersion: typeof AI_SDK_SCHEMA_VERSIONS.executionObservation;
  readonly executionId: string;
  readonly publicationRef: string;
  readonly pipelineName: string;
  readonly pipelineVersion: string;
  readonly status: AiSdkExecutionStatus;
  readonly createdAtUtc: string;
  readonly updatedAtUtc: string;
  readonly completedAtUtc?: string;
  readonly steps: readonly AiSdkExecutionStepObservation[];
}

File: implementations/python/sdk/src/multiplexed_ai_sdk/contracts/observation/execution_observation.py
from __future__ import annotations

from dataclasses import dataclass

from ...wire_model import AiSdkWireModel
from ..common.schema_versions import AiSdkSchemaVersions
from ..executions.execution_status import AiSdkExecutionStatus
from .execution_step_observation import AiSdkExecutionStepObservation


@dataclass(frozen=True)
class AiSdkExecutionObservation(AiSdkWireModel):
    execution_id: str
    publication_ref: str
    pipeline_name: str
    pipeline_version: str
    status: AiSdkExecutionStatus
    created_at_utc: str
    updated_at_utc: str
    steps: tuple[AiSdkExecutionStepObservation, ...]
    schema_version: int = AiSdkSchemaVersions.EXECUTION_OBSERVATION
    completed_at_utc: str | None = None
